lambeq/backend/grammar.py

```diff
@@ -1211,15 +1211,24 @@ def normal_form(self, left: bool = False) -> Diagram:
             Whenever :code:`normalizer` yields the same rewrite steps
             twice.
         """
-        from lambeq.backend.snake_removal import snake_removal
-        diagram, cache = self, set()
-        for _diagram in snake_removal(diagram, left=left):
+        from lambeq.backend.snake_removal import normalize
+        diagram, cache = self.remove_snakes(left=left), set()
+        for _diagram in normalize(diagram, left=left):
             if _diagram in cache:
                 raise NotImplementedError(f'{str(self)} is not connected.')
             diagram = _diagram
             cache.add(diagram)
         return diagram
 
+    rigid_normal_form = normal_form
+
+    def remove_snakes(self, left: bool = False) -> Diagram:
```

Comment:
Thanks for this, we'll need a proper docstring here as in all other public methods, but apart of that I think it's fine.

Reply:
Thanks for pointing this out. I have included a docstring for this.